src/main/java/uk/co/ramp/event/types/CommonVirusEvent.java

```diff
@@ -8,4 +11,20 @@
   VirusStatus oldStatus();
 
   VirusStatus nextStatus();
+
+  @Value.Check
+  default void check() {
+    Preconditions.checkState(
+        checkValidTransition(),
+        "Transition from " + oldStatus() + " to " + nextStatus() + " is not valid");
```

Comment:
```suggestion
      "Transition from %s to %s is not valid", oldStatus(), nextStatus());
```

I think this is more efficient.